modules/core/src/hal_internal.cpp

```diff
@@ -100,7 +100,7 @@ template <typename fptype> static inline int
 lapack_LU(fptype* a, size_t a_step, int m, fptype* b, size_t b_step, int n, int* info)
 {
     int lda = (int)(a_step / sizeof(fptype)), sign = 0;
-    int* piv = new int[m];
+    std::vector<int> piv(m+1);
```

Comment:
Why do we need `+1` here?

If it is a bug, then this bugfix was NOT backported to 4.x.

/cc @vpisarev @asmorkalov

Reply:
maybe it's not relevant anymore. As I remember, it's just common pattern to handle m=0 case properly.